### PR TITLE
Treat maps more gently

### PR DESCRIPTION
**Pull request by treeowl**
It seems it's easy to make one sweep through the maps to generate
deltas, which avoids a lot of mess matching up keys.

### CODE REVIEW

File: compare-ticks/CompareTicks.hs [hunk 1]
```diff
@@ -54,19 +54,20 @@ deltasTable deltaType as bs (metricName, metric) cols = header ++ dataRows
              ]
     dataRows =
         map formatRow
-        $ sortBy (flip compare)
-        $ tabulateDeltas deltaType as bs metric
-    formatRow (delta, k) =
-        Cells $ [ delta', showMetric as, showMetric bs ]
-                ++ map (\(_,f) -> f k (as M.! k) (bs M.! k)) cols
+        $ sortBy (flip compare `on` (\(_id, delta, _a, _b) -> delta))
+        $ M.foldrWithKey (\k (delta, a, b) r -> (k, delta, a, b) : r) []
```
Comment (treeowl) [inline compare-ticks/CompareTicks.hs]: It'd probably be clearer to use `map (\ (k, (delta, a, b)) -> (k, delta, a, b)) . M.toList`, as ugly as that is.

File: compare-ticks/CompareTicks.hs [hunk 2]
```diff
@@ -54,19 +54,20 @@ deltasTable deltaType as bs (metricName, metric) cols = header ++ dataRows
...
+                ++ map (\(_,f) -> f k a b) cols
       where
         delta'
           | delta == 0 = "-"
           | otherwise  =
             case deltaType of
               Relative -> showSigned' (showFFloat (Just 1)) (delta*100)<>"%"
               Absolute -> showSigned' (showFFloat (Just 1)) delta
-        showMetric xs = show $ metric $ xs M.! k
+        showMetric = show . metric
```
Comment (treeowl) [inline compare-ticks/CompareTicks.hs]: Is it still worth defining `showMetric`?

Comment (bgamari) [reply compare-ticks/CompareTicks.hs]: Nope, kill it.

File: compare-ticks/CompareTicks.hs (hunk 1 above)
Comment (bgamari) [reply compare-ticks/CompareTicks.hs]: Agreed.

**Comment by treeowl** [comment]
I think this is good to go. I haven't done a full comparison, but results generally look the way I remember them.

**Pull request merged**